solutions/devsprint-leonardo-santos-1/FinanceAppTests/Modules/ActivityDetails/ActivityDetailsViewModelTests.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ActivityDetailsViewModelTests.swift
+//  FinanceAppTests
+//
+//  Created by Hoff Silva on 2022-06-23.
+//
+
+import Foundation
+import XCTest
+
+@testable import FinanceApp
+
+final class ActivityDetailsViewModelTests: XCTestCase {
+    
+    private lazy var sut: ActivityDetailsViewModel = {
+        ActivityDetailsViewModel(financeService: self.financeService)
+    }()
+    
+    private lazy var activityDetailsViewModelDelegateSpy: ActivityDetailsViewModelDelegateSpy = {
+        ActivityDetailsViewModelDelegateSpy()
+    }()
+    
+    private let financeService = FinanceService(networkClient: NetworkingClientMock())
+    
+    func test_whenCallFetchData_itShouldReturnADelegateCall() {
+        sut.delegate = activityDetailsViewModelDelegateSpy
+        sut.fetchData()
+        let expec = expectation(description: "")
+        DispatchQueue.main.async {
```

Comment:
Com o Spy do DispatchQueue acho que todos os testes passariam a ser **sincronos**, então depois que o outro PR for mergeado é só dar uma ajeitadinha ;)

Reply:
Beleza. Vou esperar entao.

Reply:
PR mergeado ;) Pode fazer as alterações

Reply:
Leo, ainda nao mergeou. 
https://github.com/devpass-tech/challenge-tests-finance/pull/121

Reply:
Resolvi os conflitos da minha branch, Hoff. 😁

Reply:
😵‍💫